soccer/planning/CompositePath.hpp

```diff
@@ -0,0 +1,99 @@
+#pragma once
+#include <planning/Path.hpp>
+#include <Geometry2d/Point.hpp>
+#include <Geometry2d/Segment.hpp>
+#include <Geometry2d/CompositeShape.hpp>
+#include <Configuration.hpp>
+
+namespace Planning
+{
+	/**
+	 * @brief Represents a motion path made up of a series of Paths.
+	 *
+	 * @details The path represents a function of position given time that the robot should follow.
+	 * The path is made up of other Paths and can be made up of CompositePaths.
+	 */
+	class CompositePath: public Path
+	{
+		private:
+			//Vector of Paths
+			std::vector<std::unique_ptr<Path>> paths;
+
+			/**
+			 * Append the path to the end of the CompositePath
+			 * The path passed in should not be refenced anywhere else.
+			 */
+			void append(Path *path);
+
+			//Saving some variables to speed up computation
+			float duration;
+
+		public:
+			/** default path is empty */
+			CompositePath() {}
+
+			/** constructors with one path */
+			CompositePath(Path *path);
+			CompositePath(std::unique_ptr<Path> path);
+
+			/** 
+			 * Append the path to the end of the CompositePath
+			 */
+			void append(std::unique_ptr<Path> path);
+
+			/**
+			 * A path describes the position and velocity a robot should be at for a
+			 * particular time interval.  This method evalates the path at a given time and
+			 * returns the target position and velocity of the robot.
+			 *
+			 * @param[in] 	t Time (in seconds) since the robot started the path
+			 * @param[out] 	targetPosOut The position the robot would ideally be at at the given time
+			 * @param[out] 	targetVelOut The target velocity of the robot at the given time
+			 * @return 		true if the path is valid at time @t, false if you've gone past the end
+			 */
+			virtual bool evaluate(float t, Geometry2d::Point &targetPosOut, Geometry2d::Point &targetVelOut) const override;
+
+			/**
+			 * Returns true if the path never touches an obstacle or additionally, when exitObstacles is true, if the path
```

Comment:
Isn't this the other way around?  I think it returns true if it *does* hit an obstacle.